rust/gui-client/src-tauri/src/client/settings.rs

```diff
@@ -140,3 +146,23 @@ pub(crate) fn load_advanced_settings() -> Result<AdvancedSettings> {
     let settings = serde_json::from_str(&text)?;
     Ok(settings)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn load_old_formats() {
```

Comment:
Making sure we don't glitch out right after an upgrade.

If the upgrade is rolled back, `serde` should ignore the extra field, but changing any setting will delete the favorites